js/interface.js

```diff
@@ -801,14 +801,35 @@
     return customMenuLoadingPromise;
   }
 
+  function loadMenuManagerTab() {
+    var data = Fliplet.Widget.getData();
+
+    if (data && data.tab) {
+      $(`.nav.nav-tabs a[href='#${data.tab}']`).trigger('click');
+    }
+  }
+
   attachObservers();
   // Load menu widgets on startup
   loadCustomMenuWidgets();
+
+  loadMenuManagerTab();
+
+  Fliplet.Studio.onMessage((eventData) => {
```

Comment:
@aahuja-r @anshukdutt You should be able to change https://github.com/Fliplet/fliplet-widget-menu-manager/blob/b6059028f9278d1b96c7eac5c4b863982daf1f8d/js/interface.js#L91 to `Fliplet.Studio.onMessage(event) => {` without breaking anything. The `window.addEventListener('message', fn)` is the old way of listening for Studio events and you can just add the following clause to https://github.com/Fliplet/fliplet-widget-menu-manager/blob/b6059028f9278d1b96c7eac5c4b863982daf1f8d/js/interface.js#L123 so that it listens to tab events triggered after the initial load, i.e. clicking on a menu when the Menu Options panel is already open.

```js
 else if (event.tab) {
  // A tab is requested
  $(`.nav.nav-tabs a[href='#${event.tab}']`).trigger('click');
}
```

Reply:
Hi @tonytlwu, 
Regarding the suggested changes 
I think I should use 
_else if (event.data && event.data.tab) {
        // A tab is requested
        $(`.nav.nav-tabs a[href='#${event.data.tab}']`).trigger('click');
      }_
because on receiving message from studio, tab will be present in event.data.
Also, Our QA confirmed all expected functionality with the above code

For, https://github.com/Fliplet/fliplet-widget-menu-manager/pull/53#discussion_r893481976 , redundant code removed.

Reply:
@anshukdutt That's fine 👌🏻